src/ConformalTracking.cc

```diff
@@ -1667,6 +1667,371 @@ void ConformalTracking::extendHighPT(UniqueKDTracks& conformalTracks, SharedKDCl
+        streamlog_out(DEBUG9) << "-- Neighbour " << neighbour << ": [x,y] = [" << results.at(neighbour)->getX() << ", "
+                              << results.at(neighbour)->getY() << "]; [subdet,layer] = ["
+                              << results.at(neighbour)->getSubdetector() << ", " << results.at(neighbour)->getLayer() << "]"
+                              << std::endl;
+
+        if (firstLayer) {  // first neighbour
+          extendInSubdet = results.at(0)->getSubdetector();
+          extendInLayer  = results.at(0)->getLayer();
+          streamlog_out(DEBUG9) << "-- firstLayer) [subdet,layer] = [" << extendInSubdet << ", " << extendInLayer << "]"
+                                << std::endl;
+          firstLayer = false;
+          break;
+        } else if (results.at(neighbour)->getSubdetector() ==
+                   hitOnCurrentLayer->getSubdetector()) {  // next layer is in same subdetector
+          if ((vertexToTracker && (results.at(neighbour)->getLayer() > hitOnCurrentLayer->getLayer())) ||
+              (!vertexToTracker && (results.at(neighbour)->getLayer() < hitOnCurrentLayer->getLayer()))) {
+            extendInSubdet = results.at(neighbour)->getSubdetector();
+            extendInLayer  = results.at(neighbour)->getLayer();
+            streamlog_out(DEBUG9) << "-- sameSub,diffLayer) [subdet,layer] = [" << extendInSubdet << ", " << extendInLayer
+                                  << "]" << std::endl;
+            break;
+          }
+        } else if ((vertexToTracker && (results.at(neighbour)->getSubdetector() > hitOnCurrentLayer->getSubdetector())) ||
+                   (!vertexToTracker && (results.at(neighbour)->getSubdetector() <
+                                         hitOnCurrentLayer->getSubdetector()))) {  // next layer is in next subdetector
+          extendInSubdet = results.at(neighbour)->getSubdetector();
+          extendInLayer  = results.at(neighbour)->getLayer();
+          streamlog_out(DEBUG9) << "-- diffSub) [subdet,layer] = [" << extendInSubdet << ", " << extendInLayer << "]"
+                                << std::endl;
+          break;
+        } else
+          continue;
+      }
+
+      // Set the condition to end the loop
+      if (extendInSubdet == final_subdet && extendInLayer == final_layer) {
+        streamlog_out(DEBUG9) << "- Ending the loop. Reached final subdet layer" << std::endl;
+        loop = false;
+      }
+
+      // Initialize variables for choosing the best neighbour in layer
+      SKDCluster bestCluster = NULL;
+      map<SKDCluster, double> bestClustersWithChi2 = {};
+      vector<SKDCluster> bestClusters = {};
+      double             bestChi2     = 0.;
+      double             chi2         = conformalTracks[currentTrack]->chi2ndof();
+      double             chi2zs       = conformalTracks[currentTrack]->chi2ndofZS();
+      UKDTrack&          tempTrack    = conformalTracks[currentTrack];
```

Comment:
right, shall we create a copy instead or just replace tempTrack with track?

Reply:
It is probably better to make a copy so that the exploratory linearRegression calls don't leak into something else, for example when no good hit is found at all?

Reply:
In the latest commit, I have addressed this point. @andresailer , please can you give me the ok?

Reply:
Yes, thanks!